### PR TITLE
Tidy up gradle stuff

### PR DESCRIPTION
**Pull request by driessamyn**
Tidy up of how dependencies are specified in root gradle, and general tidy-up.

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -16,6 +16,9 @@ platformVersion = 11
 # Artifactory
 artifactoryContextUrl = https://software.r3.com/artifactory
 
+# Gradle
+detektVersion = 1.12.0
```
Comment (benmcmahon100) [inline gradle.properties:20]: Is this used anywhere yet?

**Review by benmcmahon100 — APPROVED** [review]

File: gradle.properties (hunk 1 above)
Comment (driessamyn) [reply gradle.properties:20]: Well spotted. It isn't atm as you can't use it in the plugin block I found out. Will remove.

File: build.gradle.kts [hunk 2]
```diff
@@ -52,43 +53,78 @@ allprojects {
...
         }
 
-        withType<JavaCompile>().forEach { compileJava ->
-            compileJava.options.compilerArgs.add("-parameters")
+        withType<JavaCompile>().configureEach {
+            val compilerArgs = options.compilerArgs
+            compilerArgs.add("-parameters")
         }
 
+        // TODO: does this really need to apply to all modules or can this be moved to the modules that need it only?
```
Comment (yift-r3) [inline build.gradle.kts:105]: We do need to move this at some point.

File: build.gradle.kts [hunk 3]
```diff
@@ -52,43 +53,78 @@ allprojects {
...
+        named<JavaCompile>("compileTestJava") {
+            val compilerArgs = options.compilerArgs
+            compilerArgs.add("--add-exports")
+            compilerArgs.add("java.base/sun.security.x509=ALL-UNNAMED")
+            compilerArgs.add("--add-exports")
+            compilerArgs.add("java.base/sun.security.util=ALL-UNNAMED")
+        }
+
+        // TODO: as above, this may not apply to all modules, so maybe should be moved out
```
Comment (yift-r3) [inline build.gradle.kts:114]: This should be in all the modules.

File: build.gradle.kts [hunk 4]
```diff
@@ -52,43 +53,78 @@ allprojects {
...
+        withType<Jar>().configureEach {
+            manifest {
+                attributes("Corda-Release-Version" to packageVersion)
+                attributes("Corda-Platform-Version" to platformVersion)
+                // attributes("Corda-Revision" to revision)
```
Comment (yift-r3) [inline build.gradle.kts:119]: This need to be added to TODO (or fixed)

File: build.gradle.kts [hunk 5]
```diff
@@ -106,37 +143,12 @@ subprojects {
                     destination = file("$projectDir/build/detekt-report.xml")
                 }
                 html {
-                    enabled = true
-                    destination = file("$projectDir/build/detekt-report.html")
+                    enabled = false
```
Comment (yift-r3) [inline build.gradle.kts:146]: HTML might be useful for a local build.

File: finance/contracts/build.gradle.kts [hunk 6]
```diff
@@ -1,4 +1,7 @@
 dependencies {
+    implementation("org.jetbrains.kotlin:kotlin-stdlib-jdk8:${properties["kotlinVersion"]}")
+    implementation("org.jetbrains.kotlin:kotlin-reflect:${properties["kotlinVersion"]}")
+
```
Comment (yift-r3) [inline finance/contracts/build.gradle.kts:4]: Do we need a newline?

File: build.gradle.kts (hunk 2 above)
Comment (driessamyn) [reply build.gradle.kts:105]: Do we know exactly what needs it and/or if we still need it?

File: build.gradle.kts (hunk 4 above)
Comment (driessamyn) [reply build.gradle.kts:119]: I think we should review this when we look at publishing/packaging.

File: build.gradle.kts (hunk 5 above)
Comment (driessamyn) [reply build.gradle.kts:146]: The report didn't look useful. I thought the xml version was at least as readable (especially when opening in IDE or on command line). On CI, we use a plugin to format the xml output, so I though it's not worth the extra cost.
Strangely, the txt version was always empty (maybe a bug, but didn't think it was worth investigating)

File: build.gradle.kts (hunk 2 above)
Comment (yift-r3) [reply build.gradle.kts:105]: We know that we need it.

It's easy to check where - just remove it and see where the build fails.

File: build.gradle.kts (hunk 4 above)
Comment (yift-r3) [reply build.gradle.kts:119]: For now, let's add a TODO, so we will remember to do it.

**Review by yift-r3 — APPROVED** [review]

**Pull request merged**